crates/ruff_server/src/server/api/requests/code_action.rs

```diff
@@ -87,6 +99,36 @@ fn quick_fix(
         .collect()
 }
 
+fn noqa_comments(snapshot: &DocumentSnapshot, fixes: &[DiagnosticFix]) -> Vec<CodeActionOrCommand> {
+    fixes
+        .iter()
+        .filter_map(|fix| {
+            let edit = fix.noqa_edit.as_ref()?.clone();
```

Comment:
Nit: I'm not sure if that works but maybe
```suggestion
            let edit = fix.noqa_edit.cloned()?;
```